packages/core/src/readers/type.ts

```diff
@@ -15,7 +15,19 @@ export interface FileReader extends BaseReader {
   loadData(filePath: string, fs?: CompleteFileSystem): Promise<Document[]>;
 }
 
-// For LlamaParseReader.ts
+/**
+ * A reader takes single and multiple file paths as well as a directory Path and imports data into an array of Document objects.
+ */
+export interface MultiReader extends BaseReader {
+  loadData(filePath: string, fs?: CompleteFileSystem): Promise<Document[]>;
+  loadData(filePaths: string[], fs?: CompleteFileSystem): Promise<Document[][]>;
+  loadData(
+    directoryPath: string,
+    fs?: CompleteFileSystem,
+  ): Promise<Document[][]>;
```

Comment:
Should we support directory input?

Reply:
I'm gonna check it out again and see. I'll probably have some time in 2-3 days.

The Connect Timeout Error seems to be kind of random. Sometimes it works as expected, sometimes it times out, so I'll recheck everything anyway.

Reply:
yes, the SimpleDirectoryReader already supports loading multiple files from a directory.

I think we have three concerns:
1. Walking the filesystem (already done by the SimpleDirectoryReader)
2. Reading a single file (done by the reader)
3. Loading files in parallel (new concern in this PR)

If we handle this new concern by a single reader (here LlamaParseReader), then we can only use it with this reader. 

How about adding `numWorkers` to SimpleDirectoryReader instead?

@InsightByAI if you like you can split your PR in two, the first one just adding the missing filetypes

Reply:
@marcusschiesser The issue for me with adding `numWorkers` to SimpleDirectoryReader was that it would load in batches just fine, but it would load the next batch right after finishing the 1st without waiting for the parse results.

I think the options would be to either:
1. have SimpleDirectoryReader wait for a finish signal from the Parser
2. have SimpleDirectoryReader pass on a list of all filePaths and let the parallel loading be handled by the Parser

I concentrated only on LlamaParse because I think the requirements for `numWorkers` in most other readers is quite different, since in LlamaParse the "workers" kind of refers to the max amount of files the LlamaParse API can accept at once, which is limited to 10.

If I have powerful enough hardware, I could probably use a 100 or more "workers" to load a 100 files parallel with PDFReader. 
Not sure if a `numWorkers` that limits one reader to 10 workers and has to wait for results from an API endpoint, while allowing another one to use unlimited workers and only be limited by hardware speed would be easy to implement/work that well? 

I'm going to check how python implements this. The Llama-Parse standalone can accept arrays of files but not a directory afaik, so I'll check how that works with an equivalent llama-index python directory Reader.

I'll split the PR later. Thanks for the elaborate review!

Reply:
Can you re-add `numWorkers` to SimpleDirectorReader and add a test case for the problem? Then we can have a look at it